apps/web/lib/checkForEmptyAssignment.ts

```diff
@@ -0,0 +1,35 @@
+import type { SchedulingType } from "@calcom/prisma/enums";
+
+import type { EventTypeAssignedUsers, EventTypeHosts } from "~/event-types/views/event-types-single-view";
+
+// This function checks if EventType requires assignment.
+// returns true: if EventType requires assignment but there is no assignment yet done by the user.
+// returns false: for all other scenarios.
+export function checkForEmptyAssignment({
+  assignedUsers,
+  hosts,
+  schedulingType,
+  assignAllTeamMembers,
+}: {
+  assignedUsers: EventTypeAssignedUsers;
+  hosts: EventTypeHosts;
+  schedulingType: SchedulingType | null;
+  assignAllTeamMembers: boolean;
+}): boolean {
+  // If Team-events have assignAllTeamMembers checked, return false as assignemnt is complete.
+  if (assignAllTeamMembers) {
+    return false;
+  }
+
+  // For Managed EventType, check if assigned users are empty.
+  if (schedulingType === "MANAGED" && assignedUsers.length === 0) {
+    return true;
+  }
+
+  // For Non-Managed EventType (i.e RoundRobin or Collective), check if hosts are empty.
+  if ((schedulingType === "ROUND_ROBIN" || schedulingType === "COLLECTIVE") && hosts.length === 0) {
+    return true;
+  }
```

Comment:
```
if (isManagedEventType ? assignedUsers.length === 0 : hosts.length === 0)) {
    return true;
  }
```
This simply remove all the extra work and import. I donno why you changed it

Reply:
Both are doing exactly the same, i thought this will be more clearer and readable / understandable, because there was some confusion with ManagedEventType earlier.

Reply:
Its the same earlier the check if SchedulingType === 'MANAGED' was done before calling the function, now SchedulingType is passed as argument. Also if new SchedulingType is added in future, we can still handle this inside this fn only

Reply:
Yes, I know. However, it would be simpler. If we add the new SchedulingType in the future; we would need to update not only here but also in most of the places. PLs do it.